docs/earthfile/earthfile.md

```diff
@@ -737,6 +737,35 @@ The `<context-path>` is the path where the Dockerfile build context exists. By d
 
 Specify an alternative Dockerfile to use. The `<dockerfile-path>` can be either a path on the host system, relative to the current Earthfile, or an [artifact reference](../guides/target-ref.md#artifact-reference) pointing to a Dockerfile.
 
+{% hint style='info' %}
+It is possible to split the `Dockerfile` and the build context across two seperate [artifact references](../guides/target-ref.md#artifact-reference):
+
+```Earthly
+FROM alpine
+
+mybuildcontext:
+    WORKDIR /mydata
+    RUN echo mydata > myfile
+    SAVE ARTIFACT /mydata
+
+mydockerfile:
+    RUN echo "
+FROM busybox
+COPY myfile .
+RUN cat myfile" > Dockerfile
+    SAVE ARTIFACT Dockerfile
+
+docker:
+    FROM DOCKERFILE -f +mydockerfile/Dockerfile +mybuildcontext/mydata/*
+    SAVE IMAGE testimg:latest
+```
+
+Alternatively, if both the `Dockerfile` and build context are inside the same target, one must reference the same target twice, e.g. `FROM DOCKERFILE -f +target/dir/Dockerfile +target/dir`.
+
+Other specifications may currently face issues as documented in [#1410](https://github.com/earthly/earthly/issues/1410).
```

Comment:
I would perhaps just say that `+mybuildcontext/mydata` (copies the whole dir, including the wrapping dir `mydata`) is not enough and that `/*` is necessary (copies the contents of the dir only), rather than reference this issue.